### PR TITLE
Group repo dependencies into a macro.

### PR DESCRIPTION
**Pull request by xingao267**
Also fix a bug when autoconfig target is located in the root BUILD file
of the workspace.

### CODE REVIEW

File: rules/docker_config.sh.tpl [hunk 1]
```diff
@@ -23,7 +23,7 @@ main() {
   # Expand a tar file with the repo if needed
   %{EXPAND_REPO_CMD}
   # Call the script produced by the docker_build rule to load the image
-  %{LOAD_IMAGE_SH}
+  ./%{LOAD_IMAGE_SH}
```
Comment (erain) [inline rules/docker_config.sh.tpl]: Do we want to include this change in this PR?

File: WORKSPACE [hunk 2]
```diff
@@ -36,6 +32,22 @@ load(
 
 container_repositories()
 
+load("@io_bazel_rules_go//go:def.bzl", "go_register_toolchains", "go_rules_dependencies")
+
+go_rules_dependencies()
```
Comment (erain) [inline WORKSPACE]: It is a pity that we couldn't call this repository rule in our repository rule... Maybe we can update README to indicate how should import bazel-toolchains repo in WORKSPACE as well?

Comment (xingao267) [reply WORKSPACE]: Yea. good idea. I forgot to do that. Will do.

File: rules/docker_config.sh.tpl (hunk 1 above)
Comment (xingao267) [reply rules/docker_config.sh.tpl]: I'll undo this.

File: WORKSPACE (hunk 2 above)
Comment (xingao267) [reply WORKSPACE]: Updated the instructions in the docker_config.bzl, which is where the instructions located before.

**Pull request merged**